src/App.tsx

```diff
@@ -1,61 +1,122 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+const findUserById = (id: number) => {
+  return usersFromServer.find(user => user.id === id) || null;
+};
+
+const todosWithUsers = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const [newTodos, setNewTodos] = useState<Todo[]>(todosWithUsers);
+
+  const handleTitleInput = ((e: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(e.target.value);
+    setHasTitleError(false);
+  });
+
+  const handleUserSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+e.target.value);
+    setHasUserError(false);
+  };
+
+  const resetForm = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = ((e: React.ChangeEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title && !userId) {
+      setHasTitleError(!hasTitleError);
+      setHasUserError(!hasUserError);
+
+      return;
+    }
+
+    if (!title) {
+      setHasTitleError(!hasTitleError);
+
+      return;
+    }
+
+    if (!userId) {
+      setHasUserError(!hasUserError);
+
+      return;
+    }
```

Comment:
you actually don't need these

Reply:
I need this to prevent changes and raise individual errors if there are any